cmd/agent/agent_test.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+func TestKubeletCertificateNeedsRegeneration(t *testing.T) {
+	testCertFileLocation := "testdata/kube-kubelet-167-71-188-113.pem"
+	testCertKeyFileLocation := "testdata/kube-kubelet-167-71-188-113-key.pem"
+
+	type testcase struct {
+		testCertLocation    string
+		testCertKeyLocation string
+		testIPAddress       string
+		testHostName        string
+		testTime            time.Time
+		want                bool
+	}
+
+	testCases := []testcase{
+		{
+			testCertLocation:    testCertFileLocation,
+			testCertKeyLocation: testCertKeyFileLocation,
+			testIPAddress:       "167.71.188.113",
+			testHostName:        "cert-test1",
+			testTime:            time.Date(2022, 11, 3, 12, 1, 1, 1, time.Local),
+			want:                false,
+		},
+		{
+			testCertLocation:    testCertFileLocation,
+			testCertKeyLocation: testCertKeyFileLocation,
+			testIPAddress:       "192.168.1.1",
+			testHostName:        "cert-test1",
+			testTime:            time.Date(2022, 11, 3, 12, 1, 1, 1, time.Local),
+			want:                true,
+		},
+		{
+			testCertLocation:    testCertFileLocation,
+			testCertKeyLocation: testCertKeyFileLocation,
+			testIPAddress:       "167.71.188.113",
+			testHostName:        "bad-hostname",
+			testTime:            time.Date(2022, 11, 3, 12, 1, 1, 1, time.Local),
+			want:                true,
+		},
+		{
+			testCertLocation:    testCertFileLocation,
+			testCertKeyLocation: testCertKeyFileLocation,
+			testIPAddress:       "167.71.188.113",
+			testHostName:        "cert-test1",
+			testTime:            time.Date(2032, 10, 30, 12, 1, 1, 1, time.Local), // ~ 30 hours until expiry
+			want:                true,
+		},
+	}
+
+	for _, c := range testCases {
+		got, err := KubeletCertificateNeedsRegeneration(c.testIPAddress, c.testHostName, c.testCertLocation, c.testCertKeyLocation, c.testTime)
+		assert.Equal(t, nil, err)
+		assert.Equal(t, c.want, got)
+	}
+}
```

Comment:
```suggestion
	tests := []struct {
		name								string
		testCertLocation    string
		testCertKeyLocation string
		testIPAddress       string
		testHostName        string
		testTime            time.Time
		want                bool
	}{
		{
			name:								 "valid cert",
			testCertLocation:    testCertFileLocation,
			testCertKeyLocation: testCertKeyFileLocation,
			testIPAddress:       "167.71.188.113",
			testHostName:        "cert-test1",
			testTime:            time.Date(2022, 11, 3, 12, 1, 1, 1, time.Local),
			want:                false,
		},
		{
			name:								 "bad ip",
			testCertLocation:    testCertFileLocation,
			testCertKeyLocation: testCertKeyFileLocation,
			testIPAddress:       "192.168.1.1",
			testHostName:        "cert-test1",
			testTime:            time.Date(2022, 11, 3, 12, 1, 1, 1, time.Local),
			want:                true,
		},
		{
			name:								 "bad hostname",
			testCertLocation:    testCertFileLocation,
			testCertKeyLocation: testCertKeyFileLocation,
			testIPAddress:       "167.71.188.113",
			testHostName:        "bad-hostname",
			testTime:            time.Date(2022, 11, 3, 12, 1, 1, 1, time.Local),
			want:                true,
		},
		{
			name:								 "30 hours until expired",
			testCertLocation:    testCertFileLocation,
			testCertKeyLocation: testCertKeyFileLocation,
			testIPAddress:       "167.71.188.113",
			testHostName:        "cert-test1",
			testTime:            time.Date(2032, 10, 30, 12, 1, 1, 1, time.Local),
			want:                true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got, err := KubeletCertificateNeedsRegeneration(tt.testIPAddress, tt.testHostName, tt.testCertLocation, tt.testCertKeyLocation, tt.testTime)
			assert.Equal(t, nil, err)
			assert.Equal(t, tt.want, got)
		}
	}
}
```
This is usually the more idiomatic way to write tests, if you are using goland you should see each entry in `tests` have a green arrow next to it so you can run the tests individually. I think it has to be both an anonymous struct definition as well as wrapped in a `t.Run()`, as I just tried it with a `testcase` struct and it didn't give me any test prompts.

Reply:
I wish there was a way to ignore whitespace when making a suggestion.

Reply:
Good to know, test is updated to use `t.Run` so we can include test case names